shared/debrid.py

```diff
@@ -526,3 +536,168 @@ class TorboxTorrent(Torbox, Torrent):
+                    if usenet['id'] == self.id:
+                        usenet['status'] = self._normalize_status(usenet['download_state'], usenet['download_finished'])
+                        self._info = usenet
+                        return self._info
+                
+                await asyncio.sleep(1)
+        return self._info
+
+    async def selectFiles(self):
+        pass
+
+    def delete(self):
+        self._enforceId()
+
+        deleteRequest = retryRequest(
+            lambda: requests.delete(urljoin(torbox['host'], "usenet/controlusenetdownload"), headers=self.headers, data={'usenet_id': self.id, 'operation': "delete"}),
+            print=self.print
+        )
+        return not not deleteRequest
+
+    async def getFilePath(self):
+        filename = (await self.getInfo())['name']
+
+        folderPathMountFilenameUsenet = os.path.join(self.mountTorrentsPath, filename)
+       
+        if os.path.exists(folderPathMountFilenameUsenet) and os.listdir(folderPathMountFilenameUsenet):
+            folderPathMountUsenet = folderPathMountFilenameUsenet
+        else:
+            folderPathMountUsenet = None
+
+        return folderPathMountUsenet
+
+    def _addFile(self, data=None, files=None):
+        request = retryRequest(
+            lambda: requests.post(urljoin(torbox['host'], "usenet/createusenetdownload"), headers=self.headers, data=data, files=files),
+            print=self.print
+        )
+        if request is None:
+            return None
+        
+        response = request.json()
+        self.print('response info:', response)
+        
+        if response.get('detail') == 'queued':
+            return None
+        
+        self.id = response['data']['usenetdownload_id']
+
+        return self.id
+    
+    def _addTorrentFile(self):
+        raise Exception("Not Implemented")
+
+    def _addMagnetFile(self):
+        raise Exception("Not Implemented")
+
+    def _addNZBFile(self):
+        nameusenet = self.f.name.split('/')[-1]
+        files = {'file': (nameusenet, self.f, 'application/octet-stream')}
+        return self._addFile(files=files)
+
+    def _normalize_status(self, status, download_finished):
+        if download_finished:
+            return self.STATUS_COMPLETED
+        elif status in [
+            'completed', 'cached', 'paused', 'downloading', 'uploading',
+            'checkingResumeData', 'metaDL', 'pausedUP', 'queuedUP', 'checkingUP',
+            'forcedUP', 'allocating', 'downloading', 'metaDL', 'pausedDL',
+            'queuedDL', 'checkingDL', 'forcedDL', 'checkingResumeData', 'moving'
+        ]:
+            return self.STATUS_DOWNLOADING
+        elif status in ['error', 'stalledUP', 'stalledDL', 'stalled (no seeds)', 'missingFiles']:
+            return self.STATUS_ERROR
+        return status
+
+
+class NZB(FileBase):
+    def getHash(self):
+
+        if not self._hash:
+            self._hash = hashlib.sha1(self.fileData).hexdigest()

```

Comment:
Not sure if this is the correct hash, haven't fully tested it. Open to suggestions.